vm/src/obj/objint.rs

```diff
@@ -766,6 +770,49 @@ pub fn to_int(vm: &VirtualMachine, obj: &PyObjectRef, mut base: u32) -> PyResult
     )
 }
 
+fn str_to_int(vm: &VirtualMachine, literal: &str, mut base: u32) -> PyResult<BigInt> {
+    let mut buf = literal.chars().filter(|&c| c != '_').collect::<String>();
+
+    let is_signed = buf.starts_with('+') || buf.starts_with('-');
+    let radix_range = if is_signed { 1..3 } else { 0..2 };
+    let radix_candidate = buf.get(radix_range.clone());
+
+    // try to find base
+    if let Some(radix_candidate) = radix_candidate {
+        if let Some(matched_radix) = detect_base(&radix_candidate) {
+            if base != 0 && base != matched_radix {
+                return Err(invalid_literal(vm, literal, base));
+            } else {
+                base = matched_radix;
+            }
+            buf.drain(radix_range);
+        }
+    }
+
+    // base still not found, use default
+    if base == 0 {
+        base = 10;
```

Comment:
if base is 0 cpython doesn't allow base 10 integer starting with zero. the only acceptable case is when it is zero.
```
>>> int('0123', base=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 0: '0123'
```
however `BigInt::from_str_radix` allows. there should be check for the case